jobs/upload-brew.Jenkinsfile

```diff
@@ -0,0 +1,115 @@
+import org.yaml.snakeyaml.Yaml;
+
+node {
+    checkout scm
+    // these are script global vars
+    pipeutils = load("utils.groovy")
+    pipecfg = pipeutils.load_pipecfg()
+    libcloud = load("libcloud.groovy")
+
+}
+
+def brew_principle = pipecfg.brew.principle
```

Comment:
Mentioned this in the internal MR using this too but surfacing it here too: I think this is probably supposed to be `principal` instead?

Reply:
Also, let's document these new knobs in `docs/config.yaml`.